src/main/ts/rules/PreferFun.ts

```diff
@@ -0,0 +1,76 @@
+import { Rule } from 'eslint';
+import { ArrowFunctionExpression, Expression, FunctionExpression, Literal, Statement } from 'estree';
+import * as path from 'path';
+
+export const preferFun: Rule.RuleModule = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'Prefer using the Katamari Fun helper functions, instead of inline functions.'
+    },
+    messages: {
+      preferNoop: 'Prefer using `Fun.noop` instead of `() => {}`',
+      preferAlways: 'Prefer using `Fun.always` instead of `() => true`',
+      preferNever: 'Prefer using `Fun.never` instead of `() => false`'
+    }
+  },
+  create: (context) => {
+    const reportIfRequired = (func: FunctionExpression | ArrowFunctionExpression, literal: Literal | null) => {
+      if (literal === null) {
+        context.report({
+          node: func,
+          messageId: 'preferNoop'
+        });
+      } else if (literal.raw === 'true') {
+        context.report({
+          node: func,
+          messageId: 'preferAlways'
+        });
+      } else if (literal.raw === 'false') {
+        context.report({
+          node: func,
+          messageId: 'preferNever'
+        });
+      }
+    };
+
+    const validateFunctionBody = (func: FunctionExpression | ArrowFunctionExpression, body: Statement | Expression) => {
+      if (body.type === 'BlockStatement') {
+        const funcBody = body.body;
+        if (funcBody.length === 1) {
+          validateFunctionBody(func, funcBody[0]);
+        } else if (funcBody.length === 0) {
+          context.report({
+            node: func,
+            messageId: 'preferNoop'
+          });
+        }
+      } else if (body.type === 'ReturnStatement') {
+        if (body.argument === null || body.argument?.type === 'Literal') {
+          reportIfRequired(func, body.argument);
+        }
+      } else if (body.type === 'Literal') {
+        reportIfRequired(func, body);
+      }
+    };
+
+    // Ignore cases in the actual Katamari Fun module
+    const fileName = context.getFilename();
+    if (fileName.endsWith(path.join('katamari', 'api', 'Fun.ts')) || fileName.endsWith(path.join('katamari', 'api', 'Fun.js'))) {
```

Comment:
Only a minor thing, but maybe break these two expressions out into variables so it's clear it's `js` vs. `ts`? Took me a few seconds 😅

Reply:
I really dislike using that strategy, it just bloats things and often makes it harder to read as you have a massive block, instead of a single line. I'll try and see if we can come up with something else to improve readability.

Reply:
```typescript
new Regex(`katamari${path.sep}api${path.sep}Fun\.[tj]s`)
```

Is that gross or okay do we think?

Reply:
As discussed on slack, I've taken a partial regex approach to just remove the extension and then compare using that instead.